internalshared/configutil/http_response_headers.go

```diff
@@ -0,0 +1,186 @@
+}
+
+var ValidCustomStatusCodeCollection = []string {
+    "default",
+    "1xx",
+    "2xx",
+    "3xx",
+    "4xx",
+    "5xx",
+}
+
+const (
+    contentSecurityPolicy = "default-src 'none';   connect-src 'self';  img-src 'self' data:; script-src 'self'; style-src 'unsafe-inline' 'self'; form-action  'none'; frame-ancestors 'none'; font-src 'self'"
+    xXssProtection = "1; mode=block"
+    xFrameOptions = "Deny"
+    xContentTypeOptions = "nosniff"
+    strictTransportSecurity = "max-age=31536000; includeSubDomains"
+    contentType = "text/plain; charset=utf-8"
+)
+
+func ParseDefaultHeaders(h string) string {
+    switch h {
+    case "Content-Security-Policy":
+        return contentSecurityPolicy
+    case "X-XSS-Protection":
+        return xXssProtection
+    case "X-Frame-Options":
+        return xFrameOptions
+    case "X-Content-Type-Options":
+        return xContentTypeOptions
+    case "Strict-Transport-Security":
+        return strictTransportSecurity
+    case "Content-Type":
+        return contentType
+    default:
+        return ""
+    }
+}
+
+func setDefaultResponseHeaders(c map[string]string) map[string]string {
+    defaults := make(map[string]string)
+    // adding all parsed default headers
+    for k, v := range c {
+        defaults[k] = v
+    }
+
+    // setting all default headers that are not included in the config
+    // file under the "default" category
+    for _, hn := range DefaultHeaderNames {
+        if _, ok := c[hn]; ok {
+            continue
+        }
+        hv := ParseDefaultHeaders(hn)
+        if hv != "" {
+            defaults[hn] = hv
+        }
+    }
+
+    return defaults
+}
+
+func ParseCustomResponseHeaders(r interface{}) (map[string]map[string]string, error) {
+    if !isValidListDict(r) {
+        return nil, fmt.Errorf("response headers were not configured correctly. please make sure they're in a map")
+    }
+
+    customResponseHeader := r.([]map[string]interface{})
+    h := make(map[string]map[string]string)
+
+    for _, crh := range customResponseHeader {
+        for sc, rh := range crh {
+            if !isValidListDict(rh){
+                return nil, fmt.Errorf("invalid response header type")
+            }
+
+            if !IsValidStatusCode(sc) {
+                return nil, fmt.Errorf("invalid status code found in the config file: %v", sc)
+            }
+
+            hvl := rh.([]map[string]interface{})
+            if len(hvl) != 1 {
+                return nil, fmt.Errorf("invalid number of response headers exist")
+            }
+            hvm := hvl[0]
+            hv, err := parseHeaders(hvm)
+            if err != nil {
+                return nil, err
+            }
+
+            h[sc] = hv
+        }
+    }
+
+    // setting default custom headers
+    de := h["default"]
+    h["default"] = setDefaultResponseHeaders(de)
+
+	return h, nil
+}
+
+func isValidListDict(in interface{}) bool {
+    if _, ok := in.([]map[string]interface{}); ok {
```

Comment:
You don't really need an `if`, you can just return `ok`.  Though it's more conventional to just inline one-liners like this instead of creating a func.

Reply:
done, made inline check